src/lib/Libnet/net_server.c

```diff
@@ -528,47 +561,64 @@ wait_request(time_t waittime)
 			return (-1);
 		}
 	} else {
+		sched_processed = 0;
+		if (scks && scks->active_socks) {
+			fd_set		fdset;
+			struct timeval  timeout;
+			timeout.tv_usec = 0;
+			timeout.tv_sec = 0;
+			FD_ZERO(&fdset);
+                        for (i = 0; i < scks->active_socks; i++) {
+                                if (scks->socket_fd[i] != 0) {
+					FD_SET(scks->socket_fd[i], &fdset);
+				}
+			}
+			if (select(FD_SETSIZE, &fdset, NULL, NULL, &timeout) != -1) {
+				for (i = 0; i < scks->active_socks; i++) {
+				 	if (FD_ISSET(scks->socket_fd[i], &fdset)) {
+						log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER,
+							LOG_DEBUG, __func__, "processing priority sockets");
+						process_socket(scks->socket_fd[i]);
+						sched_processed = 1;
+					}
+				}
+
+			}
+                }
+
 		for (i = 0; i < nfds; i++) {
 			int em_fd;
 			em_fd = EM_GET_FD(events, i);
 
 #ifndef WIN32
-			/* If there is any of the following signals pending, allow a small window to handle the signal */
-			if( sigpending( &pendingsigs ) == 0) {
-				if (sigismember(&pendingsigs, SIGCHLD)
-					|| sigismember(&pendingsigs, SIGHUP)
-					|| sigismember(&pendingsigs, SIGINT)
-					|| sigismember(&pendingsigs, SIGTERM)) {
-
-					if (sigprocmask(SIG_UNBLOCK, &allsigs, NULL) == -1)
-						log_err(errno, __func__, "sigprocmask(UNBLOCK)");
-					if (sigprocmask(SIG_BLOCK, &allsigs, NULL) == -1)
-						log_err(errno, __func__, "sigprocmask(BLOCK)");
-
-					return (0);
-				}
-			}
+                        /* If there is any of the following signals pending, allow a small window to handle the signal */
+                        if( sigpending( &pendingsigs ) == 0) {
+                                if (sigismember(&pendingsigs, SIGCHLD)
+                                        || sigismember(&pendingsigs, SIGHUP)
+                                        || sigismember(&pendingsigs, SIGINT)
+                                        || sigismember(&pendingsigs, SIGTERM)) {
+
+                                        if (sigprocmask(SIG_UNBLOCK, &allsigs, NULL) == -1)
+                                                log_err(errno, __func__, "sigprocmask(UNBLOCK)");
+                                        if (sigprocmask(SIG_BLOCK, &allsigs, NULL) == -1)
+                                                log_err(errno, __func__, "sigprocmask(BLOCK)");
+
+                                        return (0);
+                                }
+                        }
 #endif
-			idx = connection_find_actual_index(em_fd);
-			if (idx < 0) {
-				return -1;
+			sched_sock = 0;
+			if (scks && scks->active_socks && sched_processed) {
+				for (j = 0; j < scks->active_socks; j++) {
+                                	if (scks->socket_fd[i] == em_fd) {
+                                        	sched_sock = 1;
```

Comment:
No need to use the variable sched_sock to find out whether the current em_fd is already processed or not. We can use the following check to get the desired result after the above loop.
if (j != scks->active_socks) ==> em_fd is already processed.
else em_fd is not processed.

Reply:
done